shared/actions/fs/platform-specific.desktop.js

```diff
@@ -350,6 +349,42 @@ const loadUserFileEdits = (state: TypedState, action) =>
     }
   })
 
+const openFilesFromWidget = (state: TypedState, {payload: {path, type}}: FsGen.OpenFilesFromWidgetPayload) => {
+  let navigation = () => []
+  if (path) {
+    const parentFolder = Types.getPathFromElements(Types.getPathElements(path).slice(0, -1))
+    navigation = () => [
+      Saga.put(
+        navigateTo([
+          Tabs.fsTab,
+          {
+            props: {path: parentFolder},
+            selected: 'folder',
+          },
+        ])
+      ),
+      Saga.put(
+        navigateAppend([
+          {
+            props: {path},
+            selected: type === 'folder' ? 'folder' : 'preview',
+          },
+        ])
+      ),
+    ]
+  } else {
+    navigation = () => [
+      Saga.put(
+        switchTo([Tabs.fsTab])
+      ),
+    ]
+  }
+  return Saga.sequentially([
+    Saga.put(ConfigGen.createShowMain()),
+    ...(navigation()),
```

Comment:
If you just write the above here directly, rather than calling `navigation()`, `openFilesFromWidget` can become an arrow function.